### PR TITLE
[Feature][Flink-K8s-V2] Add KubernetesApplicationClient.shutdown on Flink-Kubernetes-V2

### PR DESCRIPTION
**Pull request by caicancai**
Add KubernetesApplicationClient.shutdown on Flink-Kubernetes-V2

## What changes were proposed in this pull request

Add KubernetesApplicationClient.shutdown on Flink-Kubernetes-V2

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

KubernetesNativeApplicationClient.shutdown 适配到 [FlinkK8sOperator.delete](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L41);

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by caicancai** [comment]
cc @Al-assad @wolfboys

**Comment by caicancai** [comment]
This function takes into account whether the user's intention to delete Application in SP is to delete deploy at the same time. The specific method must refer to other projects, so it is currently implemented separately in the v2 module.

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientEndpoint.scala [hunk 1]
```diff
@@ -84,6 +84,10 @@ object FlinkClientEndpoint {
         if (K8sFlinkConfig.isV2Enabled) KubernetesSessionClientV2.shutdown(request)
         else KubernetesNativeSessionClient.shutdown(request)
       }
+      case KUBERNETES_NATIVE_APPLICATION => {
+        if (K8sFlinkConfig.isV2Enabled) KubernetesApplicationClientV2.shutdown(request)
+        else _
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientEndpoint.scala]: The "else _" here will return a runtime error, Modify in pattern matching to `case KUBERNETES_NATIVE_APPLICATION if K8sFlinkConfig.isV2Enabled`

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala [hunk 2]
```diff
@@ -241,4 +241,21 @@ object KubernetesApplicationClientV2 extends KubernetesClientV2Trait with Logger
...
   }
 
+  @throws[Throwable]
+  def shutdown(shutDownRequest: ShutDownRequest): ShutDownResponse = {
+    val name = shutDownRequest.clusterId
+    val namespace = shutDownRequest.kubernetesDeployParam.kubernetesNamespace
+
+    def richMsg: String => String = s"[flink-shutdown][clusterId=$name][namespace=$namespace] " + _
+
+    FlinkK8sOperator.k8sCrOpr.deleteSessionJob(namespace, name).runIOAsTry match {
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala]: It should be implemented that in KubernetesApplicationClientV2, the task of removing the Application mode is actually removing the FlinkDeployment CR, not the SessionJob CR

**Review by Al-assad — CHANGES_REQUESTED** [review]

**Comment by Al-assad** [comment]
@caicancai  It seems that some errors occurred in the compilation

**Comment by caicancai** [comment]
@Al-assad  tested

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientEndpoint.scala [hunk 3]
```diff
@@ -80,10 +80,15 @@ object FlinkClientEndpoint {
...
-        if (K8sFlinkConfig.isV2Enabled) KubernetesSessionClientV2.shutdown(request)
-        else KubernetesNativeSessionClient.shutdown(request)
-      }
+      case KUBERNETES_NATIVE_SESSION =>
+        K8sFlinkConfig.isV2Enabled match {
+          case true => KubernetesSessionClientV2.shutdown(request)
+          case _ => KubernetesNativeSessionClient.shutdown(request)
+        }
+      case KUBERNETES_NATIVE_APPLICATION =>
+        K8sFlinkConfig.isV2Enabled match {
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/FlinkClientEndpoint.scala:89]: This is still an incorrect pattern matching branch line.
When KUBERNETES_NATIVE_APPLICATION = true and K8sFlinkConfig.isV2Enables = false, a match missing error will be thrown here.

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala [hunk 4]
```diff
@@ -241,4 +242,21 @@ object KubernetesApplicationClientV2 extends KubernetesClientV2Trait with Logger
...
   }
 
+  @throws[Throwable]
+  def shutdown(shutDownRequest: ShutDownRequest): ShutDownResponse = {
+    val name = shutDownRequest.clusterId
+    val namespace = shutDownRequest.kubernetesDeployParam.kubernetesNamespace
+
+    def richMsg: String => String = s"[flink-shutdown][clusterId=$name][namespace=$namespace] " + _
+
+    FlinkK8sOperator.k8sCrOpr.deleteDeployment(namespace, name).runIOAsTry match {
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesApplicationClientV2.scala]: If you directly use the `CROperator` internal API, you need to manually untrack the corresponding resources with `FlinkK8sObserver.untrack`.

**Review by Al-assad — CHANGES_REQUESTED** [review]

**Comment by caicancai** [comment]
cc @Al-assad @wolfboys

**Comment by Al-assad** [comment]
@caicancai  This PR and PR#3216 have the same logical issue. I have fixed it in commit a993111.

**Comment by caicancai** [comment]
> @caicancai This PR and PR#3216 have the same logical issue. I have fixed it in commit [a993111](https://github.com/apache/incubator-streampark/commit/a99311195cb1d85c5575368448e964aa6f1ff117).

thank you

**Review by Al-assad — APPROVED** [review]

**Pull request merged**